src/components/ui/snackbar.svelte

```diff
@@ -1,32 +1,36 @@
 <script>
-  import { onMount, onDestroy } from 'svelte';
+  import { fly } from 'svelte/transition';
+  import dynamic from 'ui/utils/dynamic-transition.js';
 
-  export let id = 'popup';
-  export let classname = 'card';
-  export let openerclass = 'popup-open';
+  export let classname = '';
+  export let isOpen = false;
+  export let timeout = 5000;
+  export let transition = fly;
+  export let transitionOptions = { x: -20, duration: 150 };
+  let timeoutInstance = null;
 
-  export let value = false;
-  export let time = 5000;
-  $: toggleBodyClass(value);
-
-  let snackbar = null;
-  onMount(() => document.body.appendChild(snackbar));
-  onDestroy(() => document.body.removeChild(snackbar));
+  export function open() {
+    isOpen = true;
+    if (timeoutInstance != null) {
+      clearTimeout(timeoutInstance);
+    }
+    timeoutInstance = setTimeout(() => { isOpen = false; }, timeout);
```

Comment:
Or simply
```suggestion
    timeoutInstance = setTimeout(close, timeout);
```

Reply:
Nah, the `close` function also involves clearing the timeout, and it's not particularly clear why would we wanna do that, so this one looks clearer to me